### PR TITLE
Trigger node's exports.setup and exports.onRemove

### PR DESCRIPTION
**Pull request by benptc**
Ensures that Objects, Frames, and Nodes are cast from JSON data to their classes whenever they are loaded into the system. Calls the node's exports.setup function from its constructor. Adds a de-constructor to each class that should be called when deleting it, which ensures that the instance is correctly removed from the system (currently the only effect is triggering the node's onRemove function, but could be expanded in the future)

### CODE REVIEW

**Comment by benptc** [comment]
@hobinjk-ptc does this make sense to do? It adds a bit of complexity because we need to cast JSON data into the classes they represent (objects, frames, nodes) whenever they get loaded. Mostly I just want to call a node addon's `setup` function when the node is created and have the option to properly destroy it (so we can call the `onRemove` function of the node addon when it is deleted in case there's any state that needs to be stopped/reset e.g. a setInterval)

**Comment by hobinjk-ptc** [comment]
This is great! I agree that there's some potential for managing the lifecycles of objects, frames, and nodes to become a bit convoluted/complex. I think this approach is good though

File: libraries/utilities.js [hunk 1]
```diff
@@ -875,3 +883,14 @@ const getVideoDir = function (objectsPath, identityFolderName, isMobile, objectN
     return videoDir;
 }
 exports.getVideoDir = getVideoDir;
+
+/**
+ * Copies all properties from json object into the new class instance
+ * @param {Object} newInstance
+ * @param {JSON} jsonData
+ */
+exports.assignProperties = function(newInstance, jsonData) {
```
[the remark this answers is not in the document]
Comment (benptc) [reply libraries/utilities.js:892]: Oh yeah, when I started working on this there was something weird happening to prototypes being saved as properties in the recorder.js file, but that's resolved now so I agree I can get rid of this function

**Comment by benptc** [comment]
Discovered a bug where moving tools from one object to another causes the links to break (maybe the node uuid is changing?). I will fix before I convert the PR from draft.

**Pull request merged**